packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -387,7 +387,7 @@ contract SystemConfig is OwnableUpgradeable, ISemver, IGasToken {
     /// @notice Internal function for updating the fee scalars as of the Ecotone upgrade.
     /// @param _basefeeScalar     New basefeeScalar value.
     /// @param _blobbasefeeScalar New blobbasefeeScalar value.
-    function _setGasConfigEcotone(uint32 _basefeeScalar, uint32 _blobbasefeeScalar) internal {
+    function _setGasConfigEcotone(uint32 _basefeeScalar, uint32 _blobbasefeeScalar) internal virtual {
```

Comment:
If we separate out the gasconfig into two update types like I suggested, we can deprecate setGasConfigEcotone entirely.  (Of course we have make sure this one works if someone calls it, so we'll still have to override this implementation in SystemConfigHolocene.)

Reply:
Oh and you'll also want to change "setGasConfig" to virtual so we can override it to make it do the right thing. Though since it's been deprecated, you might also just make it revert if called with instructions to use the (blob)base fee setters.

Reply:
Correction to my statement above, setGasConfig() was actually not deprecated until *after* Ecotone (Ecotone didn't involve any changes to L1 contracts, which is why we had to hack in the new behavior using the old API).  The setGasConfigEcotone() was added in a later update of the L1 contracts.  IMO it should have been named setFeeScalars() --  I don't think hardfork names should be in public APIs if not entirely necessary.